osu.Game/Screens/Select/BeatmapInfoWedgeV2.cs

```diff
@@ -0,0 +1,341 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using osuTK;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Effects;
+using osu.Game.Beatmaps;
+using osu.Game.Beatmaps.Drawables;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Sprites;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Localisation;
+using osu.Game.Configuration;
+using osu.Game.Graphics.UserInterface;
+using osu.Game.Rulesets;
+using osu.Game.Rulesets.Mods;
+
+namespace osu.Game.Screens.Select
+{
+    [Cached]
+    public partial class BeatmapInfoWedgeV2 : VisibilityContainer
+    {
+        private const float shear_width = 21;
+        private const float wedge_height = 120;
+        private const float transition_duration = 250;
+        private const float corner_radius = 10;
+
+        /// Todo: move this const out to song select when more new design elements are implemented for the beatmap details area, since it applies to text alignment of various elements
+        private const float text_margin = 62;
+
+        private static readonly Vector2 wedged_container_shear = new Vector2(shear_width / wedge_height, 0);
+
+        [Resolved]
+        private IBindable<RulesetInfo> ruleset { get; set; } = null!;
+
+        protected Container? DisplayedContent { get; private set; }
+
+        protected WedgeInfoText? Info { get; private set; }
+
+        private readonly Container difficultyColourBar;
+        private readonly StarCounter starCounter;
+
+        public BeatmapInfoWedgeV2()
+        {
+            Shear = wedged_container_shear;
+            Masking = true;
+            EdgeEffect = new EdgeEffectParameters
+            {
+                Colour = Colour4.Black.Opacity(.25f),
+                Type = EdgeEffectType.Shadow,
+                Radius = corner_radius,
+                Roundness = corner_radius
+            };
+            CornerRadius = corner_radius;
+
+            Children = new Drawable[]
+            {
+                // These elements can't be grouped with the rest of the content, due to being present either outside or under the backgrounds area
+                difficultyColourBar = new Container
+                {
+                    Colour = Colour4.Transparent,
+                    Depth = float.MaxValue,
+                    Anchor = Anchor.TopRight,
+                    Origin = Anchor.TopRight,
+                    RelativeSizeAxes = Axes.Y,
+
+                    // By limiting the width we avoid this box showing up as an outline around the drawables that are on top of it.
+                    Width = 40,
```

Comment:
If we want to know what value this comes from (figma says 30 plus the corner radius workaround):
`Width = colour_bar_width + corner_radius,` or similiar variable name.